src/components/Layout/Header.tsx

```diff
@@ -1,9 +1,12 @@
 'use client';
-import { useSession, signOut } from 'next-auth/react';
+
 import Link from 'next/link';
 
+import { useAuth } from '../../context/AuthContext';
+
 export const Header: React.FC = () => {
-  const { name, email } = useSession()?.data?.user || {};
+  const { user, logout } = useAuth();
+  const { name, email } = user || {};
```

Comment:
Might be nice to make sure the user is never undefined. Imagine a bigger app where this destructuring pattern has to be repeated  bzilion times :D

Reply:
Good point!:))